mne/io/edf/edf.py

```diff
@@ -511,16 +471,547 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+
+            fid.seek(6, 1)  # 6 bytes reserved
+            equipment = np.fromstring(fid.read(8), np.uint8)
+            ip = np.fromstring(fid.read(6), np.uint8)
+            patient['headsize'] = np.fromstring(fid.read(6), np.uint16)
+            patient['headsize'] = np.asarray(patient['headsize'], np.float32)
+            patient['headsize'] = np.ma.masked_array(
+                patient['headsize'],
+                np.equal(patient['headsize'], 0), None).filled()
+            ref = np.fromstring(fid.read(12), np.float32)
+            gnd = np.fromstring(fid.read(12), np.float32)
+            n_records = np.fromstring(fid.read(8), np.int64).tolist()[0]
+
+            # record length in seconds
+            record_length = np.fromstring(fid.read(8), np.uint32)
+            if record_length[0] == 0:
+                record_length[0] = 1.
+                warn('Header information is incorrect for record length. '
+                     'Default record length set to 1.')
+
+            nchan = np.fromstring(fid.read(2), np.uint16).tolist()[0]
+            fid.seek(2, 1)  # 2bytes reserved
+
+            # Channels (variable header)
+            channels = list(range(nchan))
+            ch_names = [fid.read(16).decode().strip(' \x00')
+                        for ch in channels]
+            exclude = [ch_names.index(idx) for idx in exclude]
+
+            [fid.read(80).strip().decode() for ch in channels]
+
+            """The Physical Dimensions are encoded with a 16 bit integer
+            number, as a combination of 'unit code' + order of magnitude.
+            The list of possible codes can be found in the following
+            documents:
+            - Units codes :
+            https://sourceforge.net/p/biosig/svn/HEAD/tree/trunk/biosig/doc/units.csv
+            - Decimal factors codes:
+            https://sourceforge.net/p/biosig/svn/HEAD/tree/trunk/biosig/doc/DecimalFactors.txt
+
+            For example, the physical dimension 'uV' (microvolt) is
+            encoded as 4256 (Volt) + 19 (micro) = 4275. The code 512
+            stands for dimensionless.
+            """  # noqa
+            [fid.read(6) for ch in channels]  # phys_dim, obsolete
+            units = [np.fromstring(fid.read(2), np.uint16)[0]
+                     for ch in channels]
+            include = list()
+            for i, unit in enumerate(units):
+                if unit == 4275:  # microvolts
+                    units[i] = 1e-6
+                elif unit == 512:
+                    units[i] = 1
+                else:
+                    warn('Unknown physical dimension for channel %d. '
+                         'Assuming dimensionless.' % i)
+                    units[i] = 1
+                include.append(i)
+
+            ch_names = [ch_names[idx] for idx in include]
+            physical_min = np.array(
+                [np.fromstring(fid.read(8), np.float64)[0] for ch in channels])
```

Comment:
`np.fromfile`. And don't iterate, just read the full set of elements at once. Same with all other reads. (It will go faster and use fewer lines.)